pkg/neonvm/cpuscaling/cpuscaler.go

```diff
@@ -0,0 +1,98 @@
+package cpuscaling
```

Comment:
question: IIUC this package only works with sysfs scaling, is there a reason for that?

(asking in part because it seems like our sysfs usage could be a lot more efficient, if not for the definition of the `CPUStater` interface)

Reply:
The current package name came from the interface rather than implementation, it feels more or less natural to me to name packages after what they do, not how they do. I can rename the package, if that is not reasonable for you.

>asking in part because it seems like our sysfs usage could be a lot more efficient, if not for the definition of the CPUStater interface

I'm not sure I understand the statement. Could you please clarify? The interface is defined as a separate part that actually works with the filesystem to allow unit testing of the scaling logic.

Reply:
For some context, the reason I'm pushing on this is that I'm wary about extra complexity — we have a whole setup where we're looping through CPUs, checking their states, and keeping a running counter while we maybe update them, but we already have an easy way of finding just the CPUs we want to update — why not just use that!